File: backend/Dto/ProductionPlan/ProductionPlanDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Model;
using backend.Dto.Customer;
using backend.Dto.Order;
using backend.Dto.OrderItem;
using backend.Dto.Product;
using backend.Dto.Fabric;
using backend.Dto.WorkCenter;
using backend.Dto.ProductionPlanProduct;
using backend.Dto.ProductionPlanProductSize;
using backend.Dto.ProductionPlanStage;
using backend.Dto.Material;
using backend.Dto.BillOfMaterial;
using backend.Dto.Warehouse;
using backend.Dto.WarehouseRoom;
using backend.Dto.WarehouseShelf;
using backend.Dto.Inventory;
using backend.Dto.Outlet;
using backend.Dto.OutletDemand;
using backend.Dto.Transaction;
using backend.Dto.ActivityLog;
using backend.Model.Enums;

namespace backend.Dto.ProductionPlan
{
    public class ProductionPlanDto
    {
        // <crudgen:properties>
        public string Id { get; set; } = string.Empty;

        [Required(ErrorMessage = "PlanId is required.")]
        public string PlanId { get; set; } = string.Empty;

        public string BatchId { get; set; } = string.Empty;

        public string PlanName { get; set; } = string.Empty;

        public string DemandType { get; set; } = string.Empty;

        public string SourceId { get; set; } = string.Empty;

        public string SourceName { get; set; } = string.Empty;

        public PlanPriority Priority { get; set; }

        public PlanStatus Status { get; set; }

        public DateTime PlannedStartDate { get; set; }

        public DateTime PlannedCompletionDate { get; set; }

        public int Quantity { get; set; }

        public decimal EstimatedCost { get; set; }

        public string Supervisor { get; set; } = string.Empty;

        public string ProductionLine { get; set; } = string.Empty;

        public string MaterialWarehouse { get; set; } = string.Empty;

        public string ProductionNotes { get; set; } = string.Empty;

        public DateTime PlanDate { get; set; }

        public string OutputDestination { get; set; } = string.Empty;

        public DateTime RequiredDate { get; set; }

        public decimal Progress { get; set; }

        public bool Blocked { get; set; }

        [Required(ErrorMessage = "CreatedAt is required.")]
        public DateTime CreatedAt { get; set; }

        public string CreatedBy { get; set; } = string.Empty;

        public DateTime UpdatedAt { get; set; }

        public string UpdatedBy { get; set; } = string.Empty;

        [NotMapped]
        public List<ProductionPlanProductDto> ProductionPlanProducts { get; set; } = new List<ProductionPlanProductDto>();

        [NotMapped]
        public List<ProductionPlanStageDto> ProductionPlanStages { get; set; } = new List<ProductionPlanStageDto>();

        // </crudgen:properties>
    }
}

File: try/app/(modules)/production/dto/production-plan.dto.ts
export type PlanPriority = "Normal" | "Urgent" | "Seasonal" | "Low" | "High";
export type PlanStatus = "Draft" | "Material Check" | "Scheduled" | "In Progress" | "Completed" | "On Hold" | "Active" | "Rejected";
export type DemandType = "Customer Order" | "Outlet Replenishment" | "In-house Stock";

export interface ProductionPlanProductDto {
  id?: string;
  lineId: string;
  productId: string;
  productCode: string;
  productName: string;
  category?: string;
  variant?: string;
  quantity: number;
  requiredDate?: string;
  status: string;
  productionNotes?: string;
  sizes?: Array<{ size: string; quantity: number }>;
}

export interface ProductionPlanStageDto {
  id?: string;
  stageId: string;
  stageName: string;
  workCenter: string;
  plannedStartDate?: string;
  plannedEndDate?: string;
  status: string;
  completedQty?: number;
  rejectedQty?: number;
  leadHours?: string;
}

export interface ProductionPlanDto {
  id: string;
  planId: string;
  planNumber?: string;
  title?: string;
  batchId?: string;
  planName: string;
  demandType: DemandType;
  demandSource?: string;
  sourceId?: string;
  sourceName?: string;
  priority: PlanPriority;
  status: PlanStatus;
  plannedStartDate: string;
  plannedCompletionDate: string;
  quantity: number;
  totalQuantity?: number;
  estimatedCost?: number;
  supervisor?: string;
  productionLine?: string;
  materialWarehouse?: string;
  productionNotes?: string;
  planDate: string;
  outputDestination?: string;
  requiredDate?: string;
  startDate?: string;
  endDate?: string;
  progress?: number;
  blocked?: boolean;
  createdAt?: string;
  createdBy?: string;
  updatedAt?: string;
  updatedBy?: string;
  productionPlanProducts?: ProductionPlanProductDto[];
  productionPlanStages?: ProductionPlanStageDto[];
}

export interface ProductionSummaryDto {
  totalPlans: number;
  draftPlans: number;
  inProgressPlans: number;
  completedPlans: number;
  onHoldPlans?: number;
}
